scripts/proto_splitter/proto2multi.py

```diff
@@ -0,0 +1,219 @@
+            if 'IndexedFaceSet' in ln:
+                shapeLevel = 1
+                newProtoString = shapeLevel * indent + ' '.join(ln[-2:]) + '\n'
+                defString = ''
+                if 'DEF' in ln:
+                    defString = 'DEF ' + ln[ln.index('DEF') + 1]
+                shapeLevel = 2
+                while shapeLevel > 1:
+                    line = self.f.readline()
+                    ln = line.split()
+                    if '}' in ln or ']' in ln:
+                        shapeLevel -= 1
+                    newProtoString += shapeLevel * indent + ' '.join(ln) + '\n'
+                    if '{' in ln or '[' in ln:
+                        shapeLevel += 1
+                replaceString = self.createProto(newProtoString)
+                self.pf.write(indent * level + defString +
+                              'geometry ' + replaceString)
+                self.pf.write(indent * level + '}\n')
+            else:
+                if '}' in ln or ']' in ln:
+                    level -= 1
+                elif '{' in ln or '[' in ln:
+                    level += 1
+                self.pf.write(line)
+
+    def cleanup(self, inFile, outFile=None):
+        if inFile[-5:] == '_temp':
+            os.remove(inFile)
+        if outFile is not None:
+            os.remove(outFile)
+
+    def convert_all(self, sourcePath):
+        self.create_backup(sourcePath)
+        outPath = sourcePath
+        mkdirSafe(outPath)
+        # Find all the proto files, and store their filePaths
+        os.chdir(sourcePath)
+        # Walk the tree.
+        proto_files = []  # List of the full filepaths.
+        for root, directories, files in os.walk('./'):
+            for filename in files:
+                # Join the two strings in order to form the full filepath.
+                if filename.endswith(".proto"):
+                    filepath = os.path.join(root, filename)
+                    filepath = filepath[1:]
+                    proto_files.append(filepath)
+        for proto in proto_files:
+            inFile = sourcePath + proto
+            outFile = outPath + proto
+            print('converting ' + outFile)
+            # make a copy of our inFile, which will be read and later deleted
+            shutil.copy(inFile, inFile + '_temp')
+            inFile = inFile + '_temp'
+            self.convert(inFile, outFile)
+
+    def create_backup(self, sourcePath):
+        # Create a backup of the folder we are converting
+        backupName = os.path.basename(sourcePath) + '_backup_0'
+        backupPath = os.path.dirname(sourcePath) + '/' + backupName
+        n = 0
+        while os.path.isdir(backupPath):
+            n += 1
+            backupPath = backupPath[:-1] + str(n)
+        shutil.copytree(sourcePath,  backupPath)
+
+
+if __name__ == "__main__":
+    optParser = optparse.OptionParser(usage='usage: %prog  [options]')
+    optParser.add_option('--input', dest='inPath', default=None,
```

Comment:
You should not provide default value, but rather make this option mandatory.

Reply:
how do you make it mandatory? Is that in optparse, or just throw and error when nothing is parsed?

Reply:
Unfortunately, optparse doesn't support this, but you can indeed check and throw an error.
I am saying this because when testing I forget to add the '--input' prefix and the error I get was not really meaningful for the user.